File: client-app/src/models/article.ts
export interface Article {
  uuid: string;
  title: string;
  description: string;
  keywords: string;
  snippet: string;
  url: string;
  imageUrl: string;
  language: string;
  publishedAt: Date;
  source: string;
  category1: string;
  category2: string;
  relevanceScore: number;
  locale: string;
  topStory: boolean;
}

export type AlgoliaArticle = {
  objectID: string;
  uuid: string;
  title: string;
  description: string;
  keywords: string;
  snippet: string;
  url: string;
  imageUrl: string;
  language: string;
  publishedAt: Date;
  source: string;
  category1: string;
  category2: string;
  relevanceScore: number;
  locale: string;
  topStory: boolean;
};

File: Persistence/ScaffoldModels/Article.cs
namespace Persistence.ScaffoldModels;

public partial class Article
{
    public string Uuid { get; set; }

    public string Title { get; set; }

    public string Description { get; set; }

    public string Keywords { get; set; }

    public string Snippet { get; set; }

    public string Url { get; set; }

    public string ImageUrl { get; set; }

    public string Language { get; set; }

    public DateTime PublishedAt { get; set; }

    public string Source { get; set; }

    public string Category1 { get; set; }

    public string Category2 { get; set; }

    public double? RelevanceScore { get; set; }

    public string Locale { get; set; }

    public bool TopStory { get; set; }
}
